src/syntax/abs/mod.rs

```diff
@@ -1,6 +1,125 @@
-use crate::syntax::common::Level;
+use crate::check::monad::error::TCE;
+use crate::check::monad::TCM;
+use crate::syntax::common::{DtKind, Level, DBI};
+use crate::syntax::env::NamedEnv_;
+use crate::syntax::surf::ast::{Decl, DeclKind, Expr, Ident};
+use either::Either;
+use std::collections::BTreeMap;
 
 #[derive(Debug, Clone)]
 pub enum Abstract {
-    Type(Level),
+    Type(Ident, Level),
+    /// Bottom type
+    Bot(Ident),
+    /// Global variable
+    Var(Ident, DBI),
+    /// Local variable
+    Local(Ident, DBI),
+    /// Construct call
+    Cons(Ident, Box<Abstract>),
+    /// Apply or Pipeline in surface
+    App(Box<Self>, Box<Self>),
+    /// Dependent Type type
+    Dt(Ident, DtKind, Box<Self>, Box<Self>),
+    Pair(Ident, Box<Abstract>, Box<Abstract>),
+    Fst(Ident, Box<Abstract>),
+    Snd(Ident, Box<Abstract>),
```

Comment:
I suppose this `Ident` should be a `Location` where there's only start and end offsets.

Reply:
Nope, it's not necessary.